### ISSUE TITLE
Running parallel jobs on termux devices. 

### ISSUE THREAD

**Issue by Neo-Oli** [termux/termux-packages#225]
Hi. I'm having another issue with parallel, but it's very different than the last one (and possibly not parallel specific). That's why I opened a new issue instead of reopening the other one.  
If I want to run jobs in parallel on multiple termux devices, the commands silently fail.
**Simple Test case:**

```
$ echo "Hello\nWorld" | parallel  -Slocalhost echo
```

Yields no output.
For this to work you also need to specify the localhost port in `.ssh/config`

```
Host localhost
  Port 8022
```

and have your keyfile authorized. Basically you need to be able to run `ssh localhost` and it connects. 

All jobs sent to a remote termux machine fail without output. 

```
$ echo "Hello\nWorld" | parallel  -Stermux1,termux2 echo
```

**Expected result:**
When I run the command locally without `-S` or specifying the local device with : (`-Sserver1,server2,:,server3`) it works. 
Running locally:

```
$ echo "Hello\nWorld" | parallel echo
Hello
World
```

Running on a remote cygwin and linux server:

```
echo "Hello\nWorld" | parallel  -Scygwinserver,linuxserver echo
World
Hello
```

**What I figured out so far:**
When adding `--verbose` I can see the commands. The command for the `world` line is: 
`ssh [SERVER] -- exec perl -e \''@GNU_Parallel=("use","IPC::Open3;","use","MIME::Base64");eval"@GNU_Parallel";my$eval=decode_base64(join"",@ARGV);eval$eval;'\' JEVOVnsiUEFSQUxMRUxfUElEIn09IjI4NDQiOyRFTlZ7IlBBUkFMTEVMX1NFUSJ9PSIyIjskYmFzaGZ1bmMgPSAiIjtAQVJHVj0iZWNobyBXb3JsZCI7JHNoZWxsPSIkRU5We1NIRUxMfSI7JHRtcGRpcj0iL2RhdGEvZGF0YS9jb20udGVybXV4L2ZpbGVzL3Vzci90bXAiOyRuaWNlPTA7ZG97JEVOVntQQVJBTExFTF9UTVB9PSR0bXBkaXIuIi9wYXIiLmpvaW4iIixtYXB7KDAuLjksImEiLi4ieiIsIkEiLi4iWiIpW3JhbmQoNjIpXX0oMS4uNSk7fXdoaWxlKC1lJEVOVntQQVJBTExFTF9UTVB9KTskU0lHe0NITER9PXN1YnskZG9uZT0xO307JHBpZD1mb3JrO3VubGVzcygkcGlkKXtzZXRwZ3JwO2V2YWx7c2V0cHJpb3JpdHkoMCwwLCRuaWNlKX07ZXhlYyRzaGVsbCwiLWMiLCgkYmFzaGZ1bmMuIkBBUkdWIik7ZGllImV4ZWM6JCFcbiI7fWRveyRzPSRzPDE/MC4wMDErJHMqMS4wMzokcztzZWxlY3QodW5kZWYsdW5kZWYsdW5kZWYsJHMpO311bnRpbCgkZG9uZXx8Z2V0cHBpZD09MSk7a2lsbChTSUdIVVAsLSR7cGlkfSl1bmxlc3MkZG9uZTt3YWl0O2V4aXQoJD8mMTI3PzEyOCsoJD8mMTI3KToxKyQ/Pj44KQ==;`

When sending this command to a cygwin or linux server the output is `world`. When sending it to a remote termux machine or to localhost there is no output.
It isn't an output problem either. I've tested other commands, like `gzip` and they simply don't do anything. 
When sending a command to a mix of termux and linux servers only the termux ones fail.

**Comment by fornwall** [comment]
The `decode_base64` above leads to the following being executed:

```perl
$ENV{"PARALLEL_PID"}="2844";
$ENV{"PARALLEL_SEQ"}="2";
$bashfunc = "";
@ARGV="echo World";
$shell="$ENV{SHELL}";
$tmpdir="/data/data/com.termux/files/usr/tmp";
$nice=0;
do {
	$ENV{PARALLEL_TMP}=$tmpdir."/par".join"", map{(0..9,"a".."z","A".."Z")[rand(62)]}(1..5);
} while (-e$ENV{PARALLEL_TMP});

$SIG{CHLD}=sub{$done=1;};
$pid=fork;
unless($pid) {
	setpgrp;
	eval { setpriority(0,0,$nice) };
	exec $shell,"-c",($bashfunc."@ARGV");
	die"exec:$!\n";
}

do {
	$s=$s<1?0.001+$s*1.03:$s;
	select(undef,undef,undef,$s);
} until ($done||getppid==1);

kill(SIGHUP,-${pid}) unless $done;
wait;
exit($?&127?128+($?&127):1+$?>>8)
```

The problem seems to be that it tries to exec `$shell="$ENV{SHELL}"`, and a `SHELL` environment variable is not setup in Termux by default at the moment.

**Comment by Neo-Oli** [comment]
Thanks. With this information I was able to find a workaround in the manpage. 
Here's the relevant information:
```
       $PARALLEL_SHELL
                Use this shell the shell for the commands run by GNU Parallel:

                o $PARALLEL_SHELL. If undefined use:

                o The shell that started GNU Parallel. If that cannot be
                  determined:

                o $SHELL. If undefined use:

                o /bin/sh
```
**So this works:**
```
$ export PARALLEL_SHELL=bash
$ seq 3 | parallel -Slocalhost echo
3
1
2
```

**Comment by xeffyr** [comment]
Closing since `$SHELL` is exported by login script - https://github.com/termux/termux-packages/blob/master/packages/termux-tools/login#L11.
